ports/esp32/gccollect.c

```diff
@@ -61,7 +61,7 @@ void gc_collect(void) {
     gc_collect_end();
 }
 
-#elif CONFIG_IDF_TARGET_ESP32C3
+#elif CONFIG_IDF_TARGET_ESP32C3 || CONFIG_IDF_TARGET_ESP32C6
```

Comment:
This check should not be based on the model of the chip, but the ISA used, e.g. checking `CONFIG_IDF_TARGET_ARCH="xtensa"`

Reply:
Thanks for the suggestion, I'm not particularly familiar with Espressif IDF and didn't know these existed. 

I've updated as suggested.